### ISSUE TITLE
React Native support

### ISSUE THREAD

**Issue by enagorny** [ButterCMS/buttercms-js#92]
Official [guides](https://buttercms.com/docs/api-client/react-native/) state that React Native is fully supported.
However, following the steps in the guides doesn't work.
 ```
TypeError: AbortSignal timeout is not a function (it is undefined)
```
It seems like `AbortSignal` is not supported in the React Native js engine Hermes.

Using [v1.2.16](https://github.com/ButterCMS/buttercms-js/releases/tag/v1.2.16) doesn't sound as viable options as it's not in development anymore and likely is deprecated.

<img width="575" alt="SCR-20240729-pyiu" src="https://github.com/user-attachments/assets/9e291846-e4e3-4db1-a7a6-31b0468e4af6">

**Comment by enagorny** [comment]
@ViolanteCodes have you had a chance to check the issue?

**Comment by ViolanteCodes** [comment]
> @ViolanteCodes have you had a chance to check the issue?

@enagorny we are still looking into this (@courcelan - are you able to provide some context here?)

**Comment by courcelan** [comment]
Yes, we've investigated and it looks like there is a long-standing open issue in React Native for this missing functionality. The issue is still open over there and has not been addressed since March 2024.

We're presently looking at creating a work around to the issue.

**Comment by ViolanteCodes** [comment]
@enagorny A fix has been implemented and released under the js SDK v 2.1.1. Please let us know if you continue to have issue! Thank you :)
